src/plot.js

```diff
@@ -95,8 +95,9 @@ export function plot(options = {}) {
 
   for (const mark of marks) {
     const channels = markChannels.get(mark) ?? [];
-    const values = applyScales(channels, scales);
+    let values = applyScales(channels, scales);
     const index = filter(markIndex.get(mark), channels, values);
+    if (mark.layout != null) values = mark.layout(index, scales, values, dimensions);
```

Comment:
```suggestion
    if (mark.layout != null) values = mark.layout(index.slice(), scales, values, dimensions);
```

should we make a defensive copy?

Alternatively, if we assume that it is legal for the layout to modify the index, write it as:

```suggestion
    if (mark.layout != null) ({index, values} = mark.layout(index, scales, values, dimensions));
```

(and same in the facet call).

I think my preference would go to explicitly allowing a layout to change the index, for example a hexbin layout could group points into hexagons, and a unit chart/isotype layout could create several symbols out of (say) a bar.